### PR TITLE
Separate Game Loops

### PR DESCRIPTION
**Pull request by liamdalg**
Adds separate game loops and network event types for the client and server.

### CODE REVIEW

**Comment by RileyEv** [comment]
Think alot of the codacy stuff is garbage and can be ignored

File: ProjectFiles/Assets/Scripts/Car/DriveController.cs [hunk 1]
```diff
@@ -10,11 +10,13 @@ public class DriveController : MonoBehaviour
     public GameObject car;
     private Rigidbody rb;
     private float v;
+    public bool isControllable;
```
Comment (RileyEv) [inline ProjectFiles/Assets/Scripts/Car/DriveController.cs:14]: Add getters and setters

**Comment by ghost** [comment]
I'll fix some of the codacy stuff where it makes sense

**Comment by RileyEv** [comment]
dw I was just doing it

**Comment by liamdalg** [comment]
we can remove the behaviours now right?

**Review by RileyEv — COMMENTED** [review]
> we can remove the behaviours now right?

yeh I'll delete them now

**Comment by RileyEv** [comment]
Do we know why it wants this `Make 'gameLoops' 'readonly'.`??

**Comment by RileyEv** [comment]
Why has this got the wrong map on it?

**Comment by ghost** [comment]
> Do we know why it wants this `Make 'gameLoops' 'readonly'.`??

I suppose the actual object is never being written to, just whatever data list it points to? If setting it to readonly works we might as well I think - can be good to have stuff as private & const as possible.

**Comment by ghost** [comment]
> Why has this got the wrong map on it?

What do you mean? The scene? The correct scene is TestScene and is on there for me. 

Unless you mean the old scene should be removed - you might be right there.

**Comment by RileyEv** [comment]
> > Why has this got the wrong map on it?
> 
> What do you mean? The scene? The correct scene is TestScene and is on there for me.
> 
> Unless you mean the old scene should be removed - you might be right there.

I was being an idiot and was looking at the wrong scene...
Will delete SampleScene and Testing scene tho

**Comment by RileyEv** [comment]
urghhh rider and codacy have conflicting standards...

Function overload or optional parameters??

**Comment by liamdalg** [comment]
I think in this scenario defaults are fine since we may only want to change the IP or Port, but not both. To be honest we can just move to loading the data from a config file and avoid the defaults/overloading entirely

**Pull request merged**